### DISCUSSION TITLE
Add `proxy` fednode install mode

### DISCUSSION THREAD

**Discussion by jdogresorg** [CounterpartyXCP/CIPs#95] [CIPs]
I believe the `base` install should be the minimum amount required to run a `counterparty` instance, and not include additional components which are not necessary.

Currently `xcp-proxy` and `http-addrindexrs` are included in the fednode `base` install. These services also require the installation of `mongodb` and `redis`.

I propose that we add a new `proxy` install mode in fednode.

## Fednode Install Modes

| Mode           | Services Installed                   |
| -------------- | ------------------------------------ |
| `base` 		 | bitcoin, addrindexrs, counterparty   |
| `base_extbtc`  | addrindexrs, counterparty            |
| `proxy` 		 | `base` + xcp-proxy, http-addrindexrs |
| `counterblock` | `proxy` + counterblock               |
| `full`         | `counterblock` + counterwallet       |

**Comment by jotapea** [comment]
> I believe the base install should be the minimum amount required to run a counterparty instance

Good to hear you changed your mind.

**Comment by jotapea** [comment]
BTW, after seeing more fednode related discussions (https://github.com/CounterpartyXCP/cips/discussions/94), I think is appropriate to bring up the fact that, ideally, the protocol should be self contained in a single (minimal) application.

In other words, talks about more work on the current fednode paradigm should also consider moving away from that paradigm.

**Reply by jdogresorg** [reply]
Your comments above do not indicate one way or the other if you agree with the proposed change, and instead try to sidetrack the conversation with "paradigm" talk.

Once again, I respectfully request you keep your comments focused on the topic at hand. If you want to discuss your viewpoint of moving away from fednode to something else, please create a separate discussion for your ideas...

This is a discussion on adding a specific mode to the fednode.... not for a general discussion on shifting paradigms.

**Reply by jotapea** [reply]
Too many discussions are not great also. I think starting with a comment in another related conversation can lead to making a new discussion eventually.
